site/app/templates/polls/PollForm.twig

```diff
@@ -101,7 +100,29 @@
             value="{{ poll is not null ? poll.getReleaseDate() | date("Y-m-d") }}"
         />
         <br/>
-
+        <label for="enable-timer" class="option-title">
+            Enable Timer?
+        </label>
+        <input type="checkbox" id="enable-timer" name="enable-timer" onchange="toggleTimerInputs()" {% if poll is not null and (poll.getDuration().h != 0 or poll.getDuration().i != 0 or poll.getDuration().s != 0) %} checked {% endif %} value="enabled" />
+        <br/>
+        <div id="timer-inputs" {% if poll is null or (poll.getDuration().h == 0 and poll.getDuration().i == 0 and poll.getDuration().s == 0) %} style="display:none" {% endif %}>
+            <label class="option-title">
+                Expected Duration of Poll:
+            </label>
+            <div class="poll-input">
+                <label class="option"> Hours </label><br>
+                <input type="number" name="poll-hours" id="poll-hours" class="poll-timer" placeholder="hrs" max="24" min="0" value="{{ poll is not null ? poll.getDuration().h }}" />
+            </div>
+            <div class="poll-input">
+                <label class="option"> Minutes </label><br>
+                <input type="number" name="poll-minutes" id="poll-minutes" class="poll-timer" placeholder="min" min="0" value="{{ poll is not null ? poll.getDuration().i }}"/>
```

Comment:
Should there be a maximum value of 60 for minutes and seconds, or is this a deliberate decision to allow instructors to set "90 seconds", for example?

Reply:
This was a design choice. They should be able to choose 90 seconds. The 24 hour maximum for hours is because we are limiting duration to 24 hours.